applications/app/controllers/InteractiveController.scala

```diff
@@ -68,121 +64,90 @@ class InteractiveController(
+    val htmlResponse = () => InteractiveHtmlPage.html(page)
+    val jsonResponse = () => views.html.fragments.interactiveBody(page)
+    val res = renderFormat(htmlResponse, jsonResponse, page, Switches.all)
+    Future.successful(res)
   }
 
-  def renderDCR(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) flatMap {
-      case Left((model, blocks)) => {
-        val pageType = PageType.apply(model, request, context)
-        remoteRenderer.getInteractive(wsClient, model, blocks, pageType)
-      }
-      case Right(other) => Future.successful(RenderOtherStatus(other))
-    }
+  def renderHtml(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val pageType = PageType.apply(page, request, context)
+    remoteRenderer.getInteractive(wsClient, page, blocks, pageType)
   }
 
-  def renderDCRJson(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) map {
-      case Left((model, blocks)) => {
-        val data =
-          DotcomRenderingDataModel.forInteractive(model, blocks, request, PageType.apply(model, request, context))
-        val dataJson = DotcomRenderingDataModel.toJson(data)
-        common.renderJson(dataJson, model).as("application/json")
-      }
-      case Right(other) => RenderOtherStatus(other)
-    }
+  def renderJson(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val data =
+      DotcomRenderingDataModel.forInteractive(page, blocks, request, PageType.apply(page, request, context))
+    val dataJson = DotcomRenderingDataModel.toJson(data)
+    val res = common.renderJson(dataJson, page).as("application/json")
+    Future.successful(res)
   }
 
-  def renderDCRAmp(item: ItemResponse)(implicit request: RequestHeader): Future[Result] = {
-    itemResponseToModel(Future.successful(item)) flatMap {
-      case Left((model, blocks)) => {
-        val pageType = PageType.apply(model, request, context)
-        remoteRenderer.getAMPInteractive(wsClient, model, blocks, pageType)
-      }
-      case Right(other) => Future.successful(RenderOtherStatus(other))
-    }
-  }
-
-  def itemResponseToInteractivePickerInputData(item: ItemResponse)(implicit
-      request: RequestHeader,
-  ): Option[(DateTime, List[Tag])] = {
-    for {
-      capiDatetime <- item.content.flatMap(_.webPublicationDate)
-      datetime = DateTime.parse(capiDatetime.iso8601)
-      tags <- item.content.map(_.tags)
-    } yield (datetime, tags.toList)
+  def renderAmp(page: InteractivePage, blocks: Blocks)(implicit request: RequestHeader): Future[Result] = {
+    val pageType = PageType.apply(page, request, context)
+    remoteRenderer.getAMPInteractive(wsClient, page, blocks, pageType)
   }
 
   override def renderItem(path: String)(implicit request: RequestHeader): Future[Result] = {
```

Comment:
See the new version here. In general, I tried to separate concerns a bit, and reduce indentation (which is often a sign a method is hard to read/doing too much) via introducing an inner `render` helper method.